src/event/event.service.ts

```diff
@@ -259,6 +261,33 @@ export class EventService {
     return paginate(eventQuery, { page, limit });
   }
 
+  async searchAllEvents(
+    pagination: PaginationDto,
+    query: HomeQuery,
+  ): Promise<any> {
+    await this.getTenantSpecificEventRepository();
+    const { page, limit } = pagination;
+    const { search } = query;
+    const eventQuery = this.eventRepository
+      .createQueryBuilder('event')
+      .leftJoinAndSelect('event.user', 'user')
+      .leftJoinAndSelect('event.categories', 'categories')
+      .leftJoinAndSelect('event.group', 'group')
+      .leftJoinAndSelect('event.attendees', 'attendees')
+      .where('event.status = :status', { status: EventStatus.Published });
```

Comment:
Search should also prioritise `EventVisibility.Published` and if authorised search through `EventVisibility.Authorized` as well